ci/mage/engine.go

```diff
@@ -116,6 +135,27 @@ func (t Engine) Dev(ctx context.Context) error {
 	}
 
 	fmt.Println("export _EXPERIMENTAL_DAGGER_CLI_BIN=" + binDest)
-	fmt.Println("export _EXPERIMENTAL_DAGGER_RUNNER_HOST=docker-container://" + util.EngineContainerName)
+	fmt.Println("export _EXPERIMENTAL_DAGGER_RUNNER_HOST=docker-container://" + containerName)
+	fmt.Println("export _DAGGER_TESTS_ENGINE_TAR=" + filepath.Join(binDir, "engine.tar"))
+	fmt.Println("export PATH=" + binDir + ":$PATH")
+
 	return nil
 }
+
+// Get environment variable updates for running dagger
+func (t Engine) DevEnv(ctx context.Context) {
```

Comment:
This now lets us move a ton of the env var logic all into one place in this file, instead of needing to have loads of it in `./hack/dev` and `./hack/with-dev`.